python_transport/wirepas_gateway/transport_service.py

```diff
@@ -237,12 +243,62 @@ def _on_mqtt_wrapper_termination_cb(self):
         self.stop_dbus_client()
 
     def _set_status(self):
-        event_online = wmm.StatusEvent(self.gw_id, wmm.GatewayState.ONLINE)
+        # Create a list of different sink configs
+        configs = []
+        for sink in self.sink_manager.get_sinks():
+            config = sink.read_config()
+            if config is not None:
+                configs.append(config)
+
+        topic = TopicGenerator.make_get_configs_response_topic(self.gw_id)
+
+        event_online = wmm.StatusEvent(self.gw_id,
+                                      wmm.GatewayState.ONLINE,
+                                      sink_configs=configs,
+                                      gateway_model=self.gw_model,
+                                      gateway_version=self.gw_version,
+                                      )
 
         topic = TopicGenerator.make_status_topic(self.gw_id)
 
         self.mqtt_wrapper.publish(topic, event_online.payload, qos=1, retain=True)
 
+    def _update_status(self):
+        # First check if an update is about to be sent to avoid
+        # multiple updates in a short period (0.5s)
+        with self._status_lock:
+            if self._is_update_status_scheduled:
+                # Update is already scheduled, so our modification
+                # will be handled soon
+                logging.debug("Update already scheduled")
+                return
+            else:
+                logging.debug("Update not scheduled yet")
+                self._is_update_status_scheduled = True
+
+        def set_status_after_delay(delay_s):
+            sleep(delay_s)
+            logging.debug("Time to update the status")
+            self._is_update_status_scheduled = False
+            self._set_status()
+
+        # We are here as there was no update scheduled yet.
+        # Wait a bit
+        thread = Thread(target=set_status_after_delay, args=[0.5])
```

Comment:
Out of curiosity, I know the function is fast to execute and it doesn't change a lot but why didn't you daemonized the thread ?